src/list_tasks.js

```diff
@@ -0,0 +1,150 @@
+    spanTaskDescr.className = `idx${task.index} task_unit`;
+    const inputCheckBox = document.createElement('input');
+    inputCheckBox.className = `idx${task.index}`;
+    inputCheckBox.type = 'checkbox';
+    spanTaskDescr.appendChild(inputCheckBox);
+    inputCheckBox.addEventListener('change', (e) => {
+      if (e.target.nextSibling.classList.contains('done')) {
+        e.target.nextSibling.classList.remove('done');
+        todo.completedChange(false, parseInt(e.target.classList[0].substr(3), 10));
+      } else {
+        e.target.nextSibling.classList.add('done');
+        todo.completedChange(true, parseInt(e.target.classList[0].substr(3), 10));
+      }
+    });
+
+    const inputDescription = document.createElement('input');
+    inputDescription.type = 'text';
+    inputDescription.className = `idx${task.index} description`;
+    inputDescription.value = task.description;
+    inputDescription.readOnly = true;
+    spanTaskDescr.appendChild(inputDescription);
+    inputDescription.addEventListener('click', (e) => {
+      if (e.target.readOnly && !e.target.previousSibling.checked) {
+        const moreVert = document.querySelectorAll('.more_vert');
+        const moreIcon = document.querySelector(`.more_vert.idx${e.target.classList[0].substr(3)}`);
+        Array.from(moreVert).forEach((icon) => icon.classList.add('active'));
+        moreIcon.classList.remove('active');
+        const deleteOutline = document.querySelectorAll('.delete_outline');
+        const deleteIcon = document.querySelector(`.delete_outline.idx${e.target.classList[0].substr(3)}`);
+        Array.from(deleteOutline).forEach((icon) => icon.classList.remove('active'));
+        deleteIcon.classList.add('active');
+        const descriptionText = document.querySelectorAll('.description');
+        Array.from(descriptionText).forEach((text) => { text.readOnly = true; });
+        e.target.readOnly = false;
+      }
+    });
+    inputDescription.addEventListener('keypress', (e) => {
+      if (e.key === 'Enter') {
+        todo.update(e.target.value.trim(), parseInt(e.target.classList[0].substr(3), 10));
+        e.target.value = e.target.value.trim();
+        e.target.readOnly = true;
+        e.target.parentElement.nextSibling.classList.remove('active');
+        e.target.parentElement.nextSibling.nextSibling.classList.add('active');
+      }
+    });
+    inputDescription.addEventListener('input', (e) => {
+      if (e.target.value.trim() === '') {
+        listTasks(todo);
+        interact(todo);
+      }
+    });
+    liTaskLine.appendChild(spanTaskDescr);
+
+    const spanTrashCan = document.createElement('span');
+    spanTrashCan.className = `idx${task.index} material-icons-outlined delete_outline`;
+    spanTrashCan.innerText = 'delete_outline';
+    liTaskLine.appendChild(spanTrashCan);
+    spanTrashCan.addEventListener('click', (e) => {
+      todo.delete(parseInt(e.target.classList[0].substr(3), 10));
+      listTasks(todo);
+      interact(todo);
```

Comment:
You are invoking these two functions jointly in several parts of your code. If possible consider creating a function that invokes these two methods.

Reply:
The solution of creating a function that invokes these two methods doesn't work because of the linter. I can not put this new function before the listTasks function or after the listTasks function. In both cases the linter error of calling a function before defined is flagged. My solution was to call just the listTasks function, and put the interact call at the last line of code of the listTasks function.